src/common/mod.rs

```diff
@@ -73,6 +75,31 @@ impl From<Epoch> for Value {
     }
 }
 
+type SetL1BlockValueInput = (u64, u64, U256, FixedBytes, u64, FixedBytes, U256, U256);
+
+/// Minimal L1Block ABI from the contract [implementation](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/L2/L1Block.sol).
+const L1_BLOCK_CONTRACT_ABI: &str = r#"[
+                function setL1BlockValues(uint64 _number,uint64 _timestamp, uint256 _basefee, bytes32 _hash,uint64 _sequenceNumber,bytes32 _batcherHash,uint256 _l1FeeOverhead,uint256 _l1FeeScalar) external
+            ]"#;
+
+impl TryFrom<Bytes> for Epoch {
+    type Error = eyre::Report;
+
+    /// Performs the conversion from the `setL1BlockValues` calldata to [Epoch].
+    fn try_from(tx_calldata: Bytes) -> std::result::Result<Self, Self::Error> {
```

Comment:
You should be able to use `eyre::Result` here (or just `Result` since it is already imported) if I remember correctly since the `Error` type is `eyre::Report`.

Reply:
Oh yes, my bad for forgetting to update it 😅